iree/tools/utils/cpu_features.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_TOOLS_UTILS_CPU_FEATURES_H_
+#define IREE_TOOLS_UTILS_CPU_FEATURES_H_
```

Comment:
Do you see this moving into [`iree/base/internal`](https://github.com/google/iree/tree/main/iree/base/internal) at some point, perhaps replacing third_party/cpuinfo? The existing code in https://github.com/google/iree/tree/main/iree/tools/utils is all higher level helpers built on the lower level HAL/VM/base primitives, while this code looks closer to [fpu_state.h](https://github.com/google/iree/blob/main/iree/base/internal/fpu_state.h), [prng.h](https://github.com/google/iree/blob/main/iree/base/internal/prng.h), [atomics.h](https://github.com/google/iree/blob/main/iree/base/internal/atomics.h), and other code built on top of _operating system_ or _math_ primitives.

(combined with my other comments on this PR... we shouldn't be adding large chunks of features work into "utils" directories throughout the tree - at that point they deserve a more dedicated location)

Reply:
happy to move this anywhere deemed appropriate :-)

Reply:
OK. For now it's only used by one tool in `tools/`. I'll keep this in mind if/when it gets used outside of it (I admit it's the hope/intent, so this was a useful point to raise :-) )

Reply:
tools/ is good for now - we'll want something different for runtime use and keeping this more tightly scoped will make it easier to yank out at some point (or always keep but for these different purposes)